Automattic/wp-calypso
# Admin Menu - add skeleton of data layer and state

This PR adds new `admin-menu` state and a data-layer implementation (on sites with the correct `nav-unification` feature flag _only_) to fetch data from the new REST API endpoint proposed in D49005-code.

There is no UI at this time and all testing must be performed via the browser console.

Please note that ultimately this state will end up being keyed by `siteId` but that will be handled in a followup PR based on [this Issue](https://github.com/Automattic/wp-calypso/issues/45511).

Fixes https://github.com/Automattic/wp-calypso/issues/45465
Fixes https://github.com/Automattic/wp-calypso/issues/45464


#### Testing instructions

* Apply D49005-code to your sandbox.
* Checkout this PR.
* Build Calypso `yarn && yarn start`.
* Pick any site and load the Home screen at `http://calypso.localhost:3000/home/{{YOUR_SITE_HERE}}`.
* Open browser dev tools console.
* Type `state.adminMenu` - you should see `{}` logged to console to indicate the empty default state.
* Type `dispatch({ type: 'ADMIN_MENU_REQUEST', siteId: 123456});` which will trigger the API request for the admin menu data - you should see no errors and `undefined` returned in console.
* Type `state.adminMenu` again - you should see the state populated with menu data with the `123456` as the key in the object where the menu data is stored.
* Type `dispatch({ type: 'ADMIN_MENU_REQUEST', siteId: 56789});` which will trigger another API request for the admin menu data but this time with a different siteId - you should see no errors and `undefined` returned in console.
* Type `state.adminMenu` again - you should see the state populated with menu data under keys for `123456` and `56789` respectively. This indicates the keyed reducer is working.

Now also check that this state is not available for sites without the correct `nav-unification` flag:

* Append `?flags=-nav-unification` to your browser URL.
* Reload.
* Type `state.adminMenu` in console again - you should see `undefined` logged as the state will not have been registered.
